app/javascript/src/Onboarding.jsx

```diff
@@ -191,42 +196,48 @@ class Onboarding extends Component {
   }
 
   handleCheckAllUsers() {
-    if (this.state.checkedUsers.length < this.state.users.length) {
-      this.setState({ checkedUsers: this.state.users.slice() });
-    } else {
-      this.setState({ checkedUsers: [] });
+    const { users, checkedUsers: prevCheckedUsers } = this.state;
+    let checkedUsers = [];
+
+    if (prevCheckedUsers.length < users.length) {
+      checkedUsers = users.slice();
     }
+    this.setState({ checkedUsers });
   }
 
   handleProfileChange(event) {
-    let newProfileInfo = this.state.profileInfo;
-    newProfileInfo[event.target.name] = event.target.value;
-    this.setState({profileInfo: newProfileInfo})
+    const { name, value } = event.target;
+
+    this.setState(prevState => ({
+      profileInfo: {
+        ...prevState.profileInfo,
+        [name]: value,
+      },
+    }));
   }
 
   handleCheckUser(user) {
-    const newCheckedUsers = this.state.checkedUsers.slice();
-    if (this.state.checkedUsers.indexOf(user) > -1) {
-      const index = newCheckedUsers.indexOf(user);
-      newCheckedUsers.splice(index, 1);
-    } else {
-      newCheckedUsers.push(user);
-    }
+    const { checkedUsers } = this.state;
+    const index = checkedUsers.indexOf(user);
+    const newCheckedUsers = (index > -1) ? [...checkedUsers].splice(index, 1) : checkedUsers.concat(user);
```

Comment:
newCheckedUsers would return incorrect info here. Splice's output is what was just spliced and not the whole array.

```
let arr = [1,2,3,4,5]
let spliced = arr.splice(1,0)
// spliced = [2]
// arr = [1,3,4,5]